crypto/bn/bn_mont.c

```diff
@@ -465,3 +465,25 @@ BN_MONT_CTX *BN_MONT_CTX_set_locked(BN_MONT_CTX **pmont, CRYPTO_RWLOCK *lock,
     CRYPTO_THREAD_unlock(lock);
     return ret;
 }
+
+int ossl_bn_mont_ctx_set(BN_MONT_CTX *ctx, BIGNUM *modulus, int ri, const unsigned char *rr,
+                         size_t rrlen, uint32_t nlo, uint32_t nhi)
+{
+    if (BN_copy(&ctx->N, modulus) == NULL)
+        return 0;
+    if (BN_bin2bn(rr, rrlen, &ctx->RR) == NULL)
+        return 0;
+    ctx->ri = ri;
+#if (BN_BITS2 <=32) && defined(OPENSSL_BN_ASM_MONT)
+    ctx->n0[0] = nlo;
+    ctx->n0[1] = nhi;
+#elif BN_BITS2<=32
+    ctx->n0[0] = nlo;
+    ctx->n0[1] = 0;
+#else
+    ctx->n0[0] = ((BN_ULONG)nhi<<32)|nlo;
+    ctx->n0[1] = 0;
+#endif
+
```

Comment:
I would suggest loading the old code and looking at the BN_MONT_CTX to see what is set, to confirm ALL fields (including Ni and flags) are the same.
I would also be looking at the flags on the BIGNUM related fields to see they match also.

Reply:
Is there a prettyprint function you would recommend to dump all the flags? Worst case I can write one/I've not found an existing one yet.

Reply:
I would just breakpoint it in a debugger.